### ISSUE TITLE
When yo use a more than one AS3 declaration on cluster, error in one provides that all declaration not synced on F5.

### ISSUE THREAD

**Issue by tomaszmzyk** [F5Networks/k8s-bigip-ctlr#1937]
#### Setup Details
 CIS Version : 2.X 
 Build: f5networks/k8s-bigip-ctlr:latest       
 BIGIP Version: Big IP 15.1.3    
 AS3 Version: 3.x      
 Agent Mode: AS3      
 Orchestration: OSCP      
 Orchestration Version: 3,4        
 Pool Mode: Cluster
 Additional Setup details: OpenshiftSDN

#### Description  
Hi
We use AS3 for deployment configuration on F5.
ConfigMap is preperad for every deployment that's need to by exposed by F5.
We observed, than if you have some error/bug on one declaration, there all declaration are not synced.
How it works sync between F5 and Openshift.
Because if controller merged all declaration in one and sync with F5 this is a bug.

#### Steps To Reproduce
1) Create a few deployments for different application on Openshift
2) Create apprioprate configmaps with AS3 declaration
3) Check AS3 declaration and deploy on Openshift
4) Check logs for controller. If every declaration is OK, then sync is ok, and every application works well
5) Add some bug for one of AS3 declaration.
6) For example change replicaset for some of deployed application
7) In controllers logs you should see errors
8) When replicas done, then you should see that pods on Openshift changes to quantity that set, but on F5 there are only old pool members. Configuration not sync due te declaration error in config map that's not belong to depoyment that we changie replica set.

#### Expected Result 

When you have few declaration AS3, sync should be parallel, or controller should sync only proper declaration, declaration with errors shoul not be changed

#### Actual Result  

Whn is error in one declaration from a few, all declarations are not synced

#### Diagnostic Information
```buildoutcfg
<Configuration files, error messages, logs>
Note: Sanitize the data. For example, be mindful of IPs, ports, application names and URLs
Note: The following F5 article outlines the information required when opening an issue.
https://support.f5.com/csp/article/K60974137
```

**Comment by trinaths** [comment]
This is an expected behaviour. Created CONTCNTR-2792 for internal tracking.

**Comment by tomaszmzyk** [comment]
If this is a expected behavior, that for me is a bug. Why ?
I understand that every changes could affect a F5 cluster when the automatic sync is enabled.
Ok create a bundle from all declaration, but merge only proper declaration without this with bug, and add to logs suitable information.
In above situation declaration with errors don't have impact of all declaration and not have impact to all application.

How to validate AS3 declaration before could be deployed on Openshift, and we will be sure that don't have impact for working deployments ?

Regards

**Comment by trinaths** [comment]
@tomaszmzyk - Agree. So we created a task to handle this in way that ConfigMaps with AS3 are applied based on the Tenant not as a unified AS3 declaration. AS3 JSON can be tested using any ReST client like postman. First validate your AS3 JSON and then move to an AS3 ConfigMap.

**Comment by tomaszmzyk** [comment]
Hi
Could you please send any example for validate as3 by postman or curl.
With AS3 I'm newbie.
You request to F5 with some dry run  or different way?
Thanks in advance

**Comment by trinaths** [comment]
@tomaszmzyk - Fix coming in CIS 2.7 ETA ~ Dec 2021

**Comment by tomaszmzyk** [comment]
Hi
Thanks for information.
Regards

**Comment by trinaths** [comment]
Suggest use --filter-tenant available from CIS 2.7. Closing this issue.
